addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6728,7 +6766,43 @@ registry.ImageTools = ImageHandlerOption.extend({
      * @see this.selectClass for parameters
      */
     async setImgShape(previewMode, widgetValue, params) {
+        this.trigger_up("disable_loading_effect");
         const img = this._getImg();
+        const restoreCursor = preserveCursor(this.$target[0].ownerDocument);
+        const document = this.el.ownerDocument;
+        const isCropRequired = this._isCropRequired();
+
+        // Replace the image by its clone to avoid continues image flickering.
+        const clonedImgEl = img.cloneNode(true);
+        img.insertAdjacentElement("afterend", clonedImgEl);
+        img.classList.add("d-none");
+
+        if (previewMode === true) {
+            img.classList.add("o_we_image_cropped_preview");
+        }
```

Comment:
Do this only if we know we will crop, to limit the number of operations with the other shapes :+1: 
=> with this, it is done for every shapes